### PR TITLE
Fix GritLM kwarg

### PR DESCRIPTION
**Pull request by Muennighoff**
Not sure this is the cleanest solution but GritLM init does not take arbitrary kwargs and fails if we pass device

### CODE REVIEW

File: mteb/models/gritlm.py [hunk 1]
```diff
@@ -14,6 +14,7 @@ def gritlm_loader(**kwargs):
         raise ImportError(
             "GritLM is not installed. Please install it with `pip install gritlm`."
         )
+    kwargs.pop("device") # GritLM does automatic device placement
```
Comment (isaac-chung) [inline mteb/models/gritlm.py:17]: Could we add a check?
```suggestion
    if "device" in kwargs:
      kwargs.pop("device") # GritLM does automatic device placement
```

**Review by isaac-chung — APPROVED** [review]

File: mteb/models/gritlm.py (hunk 1 above)
Comment (Muennighoff) [reply mteb/models/gritlm.py:17]: I think kwargs.pop does not fail if the text is not in kwargs but just return None in that case, so fine without checking, no?

Comment (isaac-chung) [reply mteb/models/gritlm.py:17]: Oh right! Maybe like this?
```suggestion
    kwargs.pop("device", None) # GritLM does automatic device placement
```

I was reading [here](https://docs.python.org/3/library/stdtypes.html#dict.pop) that 
> If default is not given and key is not in the dictionary, a KeyError is raised.

Comment (Muennighoff) [reply mteb/models/gritlm.py:17]: Oh you're right; let's go with that then!

File: mteb/models/gritlm.py [hunk 2]
```diff
@@ -14,6 +14,7 @@ def gritlm_loader(**kwargs):
         raise ImportError(
             "GritLM is not installed. Please install it with `pip install gritlm`."
         )
+    kwargs.pop("device", None)  # GritLM does automatic device placement
```
Comment (KennethEnevoldsen) [inline mteb/models/gritlm.py:17]: What you could do is just remove any kwarg which isn't allowed.

Comment (Muennighoff) [reply mteb/models/gritlm.py:17]: That would require a list of kwargs that are allowed right? It's a bit complex cuz the kwargs are passed to HF AutoModel which allows lots of kwargs and may even depend on the model that is loaded I think

Comment (KennethEnevoldsen) [reply mteb/models/gritlm.py:17]: Ahh right then this fix is fine

**Pull request merged**